aws/resource_aws_dx_connection_test.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAwsDxConnection_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsDxConnectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDxConnectionConfig(acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsDxConnectionExists("aws_dx_connection.hoge"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsDxConnectionDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).dxconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_dx_connection" {
+			continue
+		}
+
+		input := &directconnect.DescribeConnectionsInput{
+			ConnectionId: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.DescribeConnections(input)
+		if err != nil {
+			return err
+		}
+		for _, v := range resp.Connections {
+			if *v.ConnectionId == rs.Primary.ID && !(*v.ConnectionState == directconnect.ConnectionStateDeleted || *v.ConnectionState == directconnect.ConnectionStateDeleting) {
```

Comment:
`directconnect.ConnectionStateDeleting` shouldn't be treated as "gone" here, IMO